### PR TITLE
Fix PGPKeygen to call key/add for public key

### PR DESCRIPTION
**Pull request by patrickxb**
First version of this engine did not call key/add unless it was pushing a private pgp key.  This fixes that.  

cc: @MarcoPolo 

r? @maxtaco

### CODE REVIEW

File: go/libkb/keyfamily.go [hunk 1]
```diff
@@ -639,6 +639,7 @@ func (kf *KeyFamily) LocalDelegate(key GenericKey) (err error) {
 // account whether the key has been cancelled at time t.
 func (ckf ComputedKeyFamily) GetKeyRoleAtTime(kid keybase1.KID, t time.Time) (ret KeyRole) {
 	if info, err := ckf.getCkiIfActiveAtTime(kid, t); err != nil {
+		ckf.G().Log.Debug("GetKeyRoleAtTime %s, %s => err %s", kid, t, err)
```
Comment (maxtaco) [inline go/libkb/keyfamily.go]: this snuck in, is it ok here?

Comment (patrickxb) [inline go/libkb/keyfamily.go]: Yeah, I think it's helpful...helped me find this issue.  But if too verbose, I can get rid of it.

File: go/engine/pgp_keygen_test.go [hunk 2]
```diff
@@ -47,3 +47,42 @@ func TestPGPKeyGen(t *testing.T) {
 		t.Errorf("result keys: %d, expected 1", len(xe.Results()))
 	}
 }
+
+func TestPGPKeyGenNoPush(t *testing.T) {
+	tc := SetupEngineTest(t, "pgpkeygen")
```
Comment (MarcoPolo) [inline go/engine/pgp_keygen_test.go]: Why not make the test above have a loop that goes through shouldPush [true, false]?

Not saying you should change this just curious

**Comment by maxtaco** [comment]
👍

File: go/engine/pgp_keygen_test.go (hunk 2 above)
Comment (patrickxb) [inline go/engine/pgp_keygen_test.go]: It was so I could just do `go test -run PGPKeyGenNoPush` repeatedly.

**Pull request merged**